frontend/src/components/teacher/dashboard/AssessmentsSection.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useMemo, useState } from "react";
+import {
+  Box,
+  Center,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+} from "@chakra-ui/react";
+
+import {
+  STATUSES,
+  type TestSessionStatus,
+} from "../../../types/TestSessionTypes";
+import { titleCase } from "../../../utils/GeneralUtils";
+import ErrorState from "../../common/info/ErrorState";
+import LoadingState from "../../common/info/LoadingState";
+import EmptySessionsMessage from "../../common/info/messages/EmptySessionsMessage";
+import TestSessionListItem from "../view-sessions/TestSessionListItem";
+import useAssessmentDataQuery from "../view-sessions/useAssessmentDataQuery";
+
+const QUERY_DATA_LIMIT_PER_STATUS = 6;
+
+// 3 tabs - this is a quick way of limiting the number of assessments
+// fetched without having complex backend logic
+const QUERY_DATA_LIMIT = QUERY_DATA_LIMIT_PER_STATUS * STATUSES.length;
```

Comment:
with this approach, we could run into the case where we get 18 active assessments which end up as 6, 0, 0 right?

Reply:
yeah... my hope was kind of that we could avoid this problem by just not running into it, but you're right I should probably fix it properly by writing a better backend. I'll make a separate pr for this tho

Reply:
i'm thinking we prob will want to sort before limit to provide some guarantee of order

Reply:
but yea separate pr